FirebaseDynamicLinks/Sources/FIRDLJavaScriptExecutor.m

```diff
@@ -73,6 +75,19 @@ - (instancetype)initWithDelegate:(id<FIRDLJavaScriptExecutorDelegate>)delegate
 
 #pragma mark - Internal methods
 - (void)start {
+// Initializing a `WKWebView` causes a memory allocation error when the process
+// is running under Rosetta translation on Apple Silicon.
+// The issue only occurs on the simulator in apps targeting below iOS 14. (Issue #7618)
+#if TARGET_OS_SIMULATOR
+  BOOL systemVersionAtLeastiOS14 = [NSProcessInfo.processInfo
+      isOperatingSystemAtLeastVersion:(NSOperatingSystemVersion){14, 0, 0}];
+  // Perform an early exit if the process is running under Rosetta translation and targeting
+  // under iOS 14.
+  if (processIsTranslated() && !systemVersionAtLeastiOS14) {
+    [self handleExecutionError:nil];
```

Comment:
Calling the delegate method in the early exit case avoids interrupting the dynamic links flow and the locale property in the delegate will be set to empty string. See [here](https://github.com/firebase/firebase-ios-sdk/blob/4662da75142fbc675e294b83415184137ec8bf79/FirebaseDynamicLinks/Sources/FIRDLDefaultRetrievalProcessV2.m#L94)